include/vcpkg/base/files.h

```diff
@@ -370,6 +409,7 @@ namespace vcpkg
 
         virtual ReadFilePointer open_for_read(const Path& file_path, std::error_code& ec) const = 0;
         ReadFilePointer open_for_read(const Path& file_path, LineInfo li) const;
+        ExpectedL<ReadFilePointer> try_open_for_read(const Path& file_path) const;
```

Comment:
I think I should prepare another change that gets rid of all non-ExpectedLs here but that's for another time.